files/lib/data/conversation/ConversationAction.class.php

```diff
@@ -310,6 +315,8 @@ public function validateMarkAsRead() {
 				$this->parameters['visitTime'] = TIME_NOW;
 			}
 		}
+
+		unset($this->parameters['userID']);
```

Comment:
Yes, the userID parameter will be unset in the validateMarkAsRead method, but will be set (if not set) in markAsRead (see link). Nevertheless, I will start a new pull request and close this one, it got too messy with the wrong commit.